### PR TITLE
Avoid copying bytes when creating JSON in /connections

### PR DESCRIPTION
**Pull request by leeavital**
### What does this PR do?

`easyjson.Marshal` will call `jwriter.Writer.BuildBytes()` which allocates
from the heap (instead of the globally shared easyjson pool)

This PR changes the connection endpoint to dump the json writer contents
straight to the http response.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: cmd/system-probe/probe.go [hunk 1]
```diff
@@ -161,14 +161,20 @@ func getClientID(req *http.Request) string {
 }
 
 func writeConnections(w http.ResponseWriter, cs *ebpf.Connections) {
-	buf, err := easyjson.Marshal(cs)
-	if err != nil {
+	jw := &jwriter.Writer{}
```
Comment (kevinconaway) [inline cmd/system-probe/probe.go]: The writer has its buffer structure which could be reused (`DumpTo` resets it)  if we can do it in a thread-safe way.

File: cmd/system-probe/probe.go [hunk 2]
```diff
@@ -161,14 +161,20 @@ func getClientID(req *http.Request) string {
...
+	cs.MarshalEasyJSON(jw)
+	if err := jw.Error; err != nil {
 		log.Errorf("unable to marshall connections into JSON: %s", err)
 		w.WriteHeader(500)
 		return
 	}
-	w.Write(buf)
-	log.Tracef("/connections: %d connections, %d bytes", len(cs.Conns), len(buf))
+	bytes, err := jw.DumpTo(w)
```
Comment (kevinconaway) [inline cmd/system-probe/probe.go]: nit, the naming here makes me think its the actual bytes returned but in reality is a count.

File: cmd/system-probe/probe.go [hunk 3]
```diff
@@ -161,14 +161,20 @@ func getClientID(req *http.Request) string {
...
+	if err != nil {
+		log.Errorf("unable to dump JSON to response")
```
Comment (kevinconaway) [inline cmd/system-probe/probe.go]: Would be good to log the error

**Review by kevinconaway — DISMISSED** [review]
Minor nits but overall 👍 

Going forward, are we still planning to use the JSON serialization once https://github.com/DataDog/datadog-agent/pull/3565 is merged?

File: cmd/system-probe/probe.go (hunk 1 above)
Comment (leeavital) [reply cmd/system-probe/probe.go]: I was originally going to do this, but after I looked at the code of  `DumpTo` I saw that all the allocated memory that stuck around lived inside the global pool, not the writer. So sharing the writer would not cut down on too many allocations.

File: cmd/system-probe/probe.go (hunk 2 above)
Comment (leeavital) [reply cmd/system-probe/probe.go]: Agreed, I changed it to `bytesWritten`

**Comment by leeavital** [comment]
@kevinconaway that's right, profobufs should replace this completely. This is just a stopgap.

File: cmd/system-probe/probe_test.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+func TestDecode(t *testing.T) {
+	rec := httptest.NewRecorder()
+
+	in := &ebpf.Connections{
+		Conns: []ebpf.ConnectionStats{
+			{
+				Source: util.AddressFromString("10.1.1.1"),
+				Dest:   util.AddressFromString("10.1.1.1"),
+				SPort:  1000,
+				DPort:  9000,
```
Comment (sfluor) [inline cmd/system-probe/probe_test.go]: nit: can we set all the fields to non-zero values ? (just to ensure that decoding worked for those) and also have different source and dests values

Comment (leeavital) [reply cmd/system-probe/probe_test.go]: Done

**Review by sfluor — APPROVED** [review]

**Pull request merged**